src/NUnitFramework/framework/Internal/TestProgressReporter.cs

```diff
@@ -11,19 +14,22 @@ namespace NUnit.Framework.Internal
     /// the async callbacks that are used to inform the client
     /// software about the progress of a test run.
     /// </summary>
-    public class TestProgressReporter : ITestListener
+    public sealed class TestProgressReporter : ITestListener
```

Comment:
I understand how sealing a class can help performance, but I'm a little cautious about this one. I think `ITestEventListener` was removed as an extension point in the framework but is still an extension point within the engine. It's plausible that someone may've extended `TestProgressReporter` in their codebase to do this.
https://docs.nunit.org/articles/nunit-engine/extensions/creating-extensions/Event-Listeners.html?q=ITestListener

@OsirisTerje you have more experience at this level, but am I reading the docs right? I'm a little confused as we also have this, but I believe this page only indicates the removal as a framework extension point: https://docs.nunit.org/articles/nunit/technical-notes/usage/Addin-Replacement-in-the-Framework.html?q=ITestListener

Reply:
Sorry for being a little late for the game. Would v4 be good time to seal (separate PR) all things that imaginable that doesn't break the current upstream (adapters, runners) and then if someone cries out just open in small point releases, opening sealed classes later isn't a breaking change but sealing too late is...